addons/controllers/packageinstallstatus_controller_test.go

```diff
@@ -0,0 +1,221 @@
+			wlcClusterBootstrap = testClusterBootstrapGet(client.ObjectKeyFromObject(wlcCluster))
+			Expect(wlcClusterBootstrap).ShouldNot(BeNil())
+			Expect(len(wlcClusterBootstrap.Status.Conditions)).Should(Equal(0))
+
+			By("verifying ClusterBootstrap 'Status.Conditions' does not get updated when PackageInstall's summarized condition is Unknown")
+			// verify for management cluster
+			testUpdatePkgInstallStatus(mngAntreaObjKey, util.UnknownCondition)
+			mngClusterBootstrap = testClusterBootstrapGet(client.ObjectKeyFromObject(mngCluster))
+			Expect(mngClusterBootstrap).ShouldNot(BeNil())
+			// Antrea is already installed into management cluster's tkg-system namespace
+			Expect(len(mngClusterBootstrap.Status.Conditions)).Should(Equal(1))
+			Expect(mngClusterBootstrap.Status.Conditions[0].Type).Should(Equal(antreaCondType))
+			// verify for workload cluster
+			testUpdatePkgInstallStatus(wlcAntreaObjKey, util.UnknownCondition)
+			testUpdatePkgInstallStatus(wlcKappObjKey, util.UnknownCondition)
+			wlcClusterBootstrap = testClusterBootstrapGet(client.ObjectKeyFromObject(wlcCluster))
+			Expect(wlcClusterBootstrap).ShouldNot(BeNil())
+			Expect(len(wlcClusterBootstrap.Status.Conditions)).Should(Equal(0))
+
+			By("verifying ClusterBootstrap 'Status.Conditions' gets updated for managed packages")
+			// verify for management cluster
+			mngClusterBootstrapStatus := testWaitForClusterBootstrapStatus(client.ObjectKeyFromObject(mngCluster), antreaCondType)
+			Expect(len(mngClusterBootstrapStatus.Conditions)).Should(Equal(1))
+			Expect(mngClusterBootstrapStatus.Conditions[0].Type).Should(Equal(antreaCondType))
+			// verify for workload cluster
+			testUpdatePkgInstallStatus(wlcAntreaObjKey, kappctrlv1alpha1.ReconcileSucceeded)
+			testUpdatePkgInstallStatus(wlcKappObjKey, kappctrlv1alpha1.Reconciling)
+			antreaCondType = "Antrea-" + runtanzuv1alpha3.ConditionType(v1alpha1.ReconcileSucceeded)
+			kappCondType := "Kapp-Controller-" + runtanzuv1alpha3.ConditionType(v1alpha1.Reconciling)
+			testWaitForClusterBootstrapStatus(client.ObjectKeyFromObject(wlcCluster), antreaCondType)
+			wlcClusterBootstrapStatus := testWaitForClusterBootstrapStatus(client.ObjectKeyFromObject(wlcCluster), kappCondType)
+			Expect(len(wlcClusterBootstrapStatus.Conditions)).Should(Equal(2))
+			Expect(wlcClusterBootstrapStatus.Conditions[0].Type).Should(Equal(antreaCondType))
+			Expect(wlcClusterBootstrapStatus.Conditions[1].Type).Should(Equal(kappCondType))
+		})
+	})
+})
+
+// testUpdatePkgInstallStatus simulates kapp controller PackageInstall status update
+func testUpdatePkgInstallStatus(objKey client.ObjectKey, appCondType kappctrlv1alpha1.AppConditionType) {
+	pkgInstall := &kapppkgiv1alpha1.PackageInstall{}
+	Eventually(func() bool {
+		if err := k8sClient.Get(ctx, objKey, pkgInstall); err != nil {
+			return false
+		}
+		condition := kappctrlv1alpha1.AppCondition{Type: appCondType, Status: corev1.ConditionTrue}
```

Comment:
question: do we want to add `Reason` and `Message` as well or we are ignoring them per design ?

Reply:
Yes per the design proposal I had shared, didn't put ```Reason``` & ```Message```, cause ```message``` in case of failure is usually something like ```message: Error (see .status.usefulErrorMessage for details)``` and not much useful. And ```reason``` is most often not set and when it is, it doesn't give much information as it is a short, machine understandable string. The most useful information of these is in ```Status.UsefulErrorMessage``` which I already fill for failures.

Reply:
These are the fields I'm setting in the ```ClusterBootstrapStatus```:

```
type Condition struct {
	// Type of condition in CamelCase
	Type ConditionType `json:"type"`

	// Status of the condition, one of True, False, Unknown
	Status corev1.ConditionStatus `json:"status"`

	// A human readable error message. Will be empty in case no error happened
	UsefulErrorMessage string `json:"usefulErrorMessage,omitempty"`

	// Last time the condition transitioned from one status to another.
	// It reflects the time any other fields in the condition changed
	LastTransitionTime metav1.Time `json:"lastTransitionTime"`
}
```